broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -0,0 +1,170 @@
+          return this.runFailed(err, {}, job, done);
+        });
+    }
+  }
+
+  static checkOperationCompletionStatus(job_data) {
+    const operationName = job_data.operation;
+    const instanceInfo = job_data.operation_details;
+    const instance_guid = instanceInfo.instance_guid;
+    const backup_guid = instanceInfo.backup_guid;
+    const deployment = instanceInfo.deployment;
+    const token = utils.encodeBase64(instanceInfo);
+    return Promise.try(() => {
+        if (operationName === 'backup') {
+          return this
+            .getFabrikClient()
+            .getInstanceBackupStatus(instanceInfo, token);
+        }
+      })
+      .then(operationStatusResponse => {
+        operationStatusResponse.jobCancelled = false;
+        operationStatusResponse.operationTimedOut = false;
+        operationStatusResponse.operationFinished = false;
+        if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
+          operationStatusResponse.operationFinished = true;
+          return operationStatusResponse;
+        } else {
+          logger.info(`Instance ${instance_guid} ${operationName} for backup guid ${backup_guid} still in-progress - `, operationStatusResponse);
+          const currentTime = new Date();
+          const backup_triggered_duration = (currentTime - new Date(instanceInfo.started_at)) / 1000;
+          return Promise
+            .try(() => bosh.director.getDirectorConfig(instanceInfo.deployment))
+            .then(directorConfig => {
+              const lock_deployment_max_duration = directorConfig.lock_deployment_max_duration;
+              if (backup_triggered_duration > lock_deployment_max_duration) {
+                //Operation timed out
+                if (!instanceInfo.abortStartTime) {
+                  //Operation not aborted. Aborting operation and with abort start time
+                  // re-registering statupoller job
+                  let abortStartTime = new Date().toISOString();
+                  instanceInfo.abortStartTime = abortStartTime;
+                  return serviceFabrikClient
+                    .abortLastBackup(_.pick(instanceInfo, ['instance_guid', 'tenant_id']))
+                    .then(() => DirectorManager.registerBnRStatusPoller(job_data, instanceInfo))
+                    .then(() => {
+                      operationStatusResponse.state = CONST.OPERATION.ABORTING;
+                      return operationStatusResponse;
+                    });
+                } else {
+                  // Operation aborted
+                  const currentTime = new Date();
+                  const abortDuration = (currentTime - new Date(instanceInfo.abortStartTime));
+                  if (abortDuration < config.backup.abort_time_out) {
+                    logger.info(`backup abort is still in progress on : ${deployment} for guid : ${backup_guid}`);
```

Comment:
can you use instance id itself in the logs here for consistency.

Reply:
Changed to deployment instead of instance id. Also job name saved as <deployment>_backup_<backup_guid>. Done keeping cross-consumption scenario in mind. As instance_guid might not be unique across platforms.

Reply:
ok , makes sense.